cabal-install/Distribution/Client/Dependency/Modular/Builder.hs

```diff
@@ -125,8 +125,13 @@ build = ana go
     go    (BS { index = _  , next = OneGoal (OpenGoal (Simple (Pkg _ _          ) _) _ ) }) =
       error "Distribution.Client.Dependency.Modular.Builder: build.go called with Pkg goal"
     go bs@(BS { index = idx, next = OneGoal (OpenGoal (Simple (Dep qpn@(Q _ pn) _) _) gr) }) =
+      -- If the package does not exist in the index, we construct an emty PChoiceF node for it
+      -- After all, we have no choices here. Alternatively, we could immediately construct
+      -- a Fail node here, but that would complicate the construction of conflict sets.
+      -- We will probably want to give this case special treatment when generating error
+      -- messages though.
       case M.lookup pn idx of
-        Nothing  -> FailF (toConflictSet (Goal (P qpn) gr)) (BuildFailureNotInIndex pn)
+        Nothing  -> PChoiceF qpn gr (P.fromList [])
```

Comment:
@grayjay But why do you think it's "more correct"? Except in the case where we have no available choices at all (the one we're talking about here), the only way we could be asked to `Fail` here is because we discovered conflicts lower in the tree. But then we're backjumping already, and we wouldn't even consider the current node if their conflict didn't include our package name.